modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -198,8 +211,9 @@ type Handler struct {
 	handleResponseSegments []*caddyfile.Dispenser
 
 	// Stores upgraded requests (hijacked connections) for proper cleanup
-	connections   map[io.ReadWriteCloser]openConnection
-	connectionsMu *sync.Mutex
+	connections           map[io.ReadWriteCloser]openConnection
+	connectionsCloseTimer **time.Timer
```

Comment:
The reason for this is the same as why there is `*sync.Mutex` and not just `sync.Mutex`. As I have found during debugging there are multiple instances of `Handler` struct floating around during runtime and I needed to synchronize whether there is `*time.Timer` among all of them or at least some of them. Specifically the registration and the cleanup code are called on different instances of `Handler`. Why is this so? Is it an unfortunately leaked reference from a non-pointer receiver method? Or does it have some other reason?

And no, although I can write some C I would not call myself a C programmer. :smile:

Reply:
They should be on the same instance. The only reason I can think of why it would be different is because some of the methods don't have pointer receivers (i.e. some methods are `(h Handler)` because they, at least currently, don't intend to modify the handler and so this is kind of a safeguard -- but that `h` will be different than if the method has a pointer receiver, i.e. `(h *Handler)`).

Reply:
Yes! Absolutely.